VAMobile/src/store/slices/authSlice.ts

```diff
@@ -171,6 +193,20 @@ export const checkFirstTimeLogin = (): AppThunk => async (dispatch) => {
   dispatch(dispatchSetFirstLogin(isFirstLogin))
 }
 
+export const checkRequestNotificationsPreferenceScreen = (): AppThunk => async (dispatch) => {
+  if (IS_TEST) {
+    // In integration tests this will change the behavior and make it inconsistent across runs
+    dispatch(dispatchSetNotificationsPreferenceScreen(false))
+    return
+  }
+
+  const setNotificationsPreferenceScreenVal = await AsyncStorage.getItem(NOTIFICATION_COMPLETED_KEY)
+  console.debug(`checkRequestNotificationPreferenceScreen: is ${!setNotificationsPreferenceScreenVal}`)
+
+  const shouldShowScreen = !setNotificationsPreferenceScreenVal
```

Comment:
I don't see any logic for resetting `NOTIFICATION_COMPLETED_KEY` at logout. If a previously logged-in user had notifications enabled, it looks like a new user signing in wouldn't see the RequestNotificationScreen?

Reply:
This is a real head scratcher for me and i'll need to bug QA about this tomorrow.

Reply:
So, the fact that this tangles with OS specific permission granting and isn't actually tied to our server side notification manager is what is selling me that we should do nothing here. This is very similar to biometric preferences. When you logout of the app we do not reset your biometric preferences, we continue to allow you to use them and if another user signed on we wouldn't re-prompt. This being the 0.1% issue is another reason I don't want it to hold us up. Thoughts?